### PR TITLE
[NU-1338] Fix: Not expected type: Null runtime error for non-nullable…

### PR DESCRIPTION
**Pull request by arkadius**
…, optional json fields that were not provided by user in sink

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: docs/Changelog.md [hunk 1]
```diff
@@ -35,6 +35,7 @@
 * [#5171](https://github.com/TouK/nussknacker/pull/5171) Breaking change: some components changed identifier - url's and identifiers in node errors are changed
 * [#5171](https://github.com/TouK/nussknacker/pull/5171) Component `mapVariable` was renamed to `record-variable`
 * [#5223](https://github.com/TouK/nussknacker/pull/5223) Legacy low level kafka components were removed
+* [#5233](https://github.com/TouK/nussknacker/pull/5233) Fix: Not expected type: Null runtime error for not-nullable, optional json fields that were not provided by user in sink
```
Comment (coutoPL) [inline docs/Changelog.md:38]: typo: non-nullable

File: engine/lite/components/kafka-tests/src/test/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaUniversalJsonFunctionalTest.scala [hunk 2]
```diff
@@ -423,6 +404,40 @@ class LiteKafkaUniversalJsonFunctionalTest
...
+    val input = KafkaConsumerRecord[String, String](cfg.sourceTopic, cfg.inputData.toString())
+
+    val result = runner.runWithStringData(jsonScenario, List(input)).validValue
+
+    result.errors shouldBe empty
+    result.success should have size 1
+    CirceUtil.decodeJsonUnsafe[Map[String, String]](result.success.head.value()) shouldBe empty
+  }
+
+  def createEditorModeScenario(config: ScenarioConfig, fieldsExpressions: Map[String, String]): CanonicalProcess = {
```
Comment (coutoPL) [inline engine/lite/components/kafka-tests/src/test/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaUniversalJsonFunctionalTest.scala:422]: can be `private`

**Review by coutoPL — APPROVED** [review]

File: utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoder.scala [hunk 3]
```diff
@@ -102,7 +102,10 @@ object BestEffortJsonSchemaEncoder {
     case ObjectField(fieldName, None, Some(_), parentSchema)
         if parentSchema.getRequiredProperties.contains(fieldName) =>
       error(s"Missing property: $fieldName for schema: $parentSchema.")
-    case ObjectField(fieldName, Some(value), Some(schema), _) =>
+    case ObjectField(fieldName, Some(value), Some(schema), parentSchema)
+        if parentSchema.getRequiredProperties.contains(fieldName) =>
+      encode(value, schema, Some(fieldName)).map(fieldName -> _)
+    case ObjectField(fieldName, Some(value), Some(schema), _) if value != null =>
```
Comment (raphaelsolarski) [inline utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoder.scala:108]: maybe we can join this case with the previous one using `||`. The outcome looks similar.

**Review by coutoPL — APPROVED** [review]
LGTM

**Review by raphaelsolarski — APPROVED** [review]

**Pull request merged**